src/GraphQL.Tests/Serialization/NewtonsoftJson/InputsConverterTests.cs

```diff
@@ -0,0 +1,302 @@
+
+            actual["values"].ShouldNotBeNull();
+            object values = actual["values"];
+            values.ShouldBeAssignableTo<IEnumerable<object>>();
+        }
+
+        [Fact]
+        public void Deserialize_ComplexValue()
+        {
+            string json = @"
+                {
+                    ""complex"": {
+                        ""int"": 123,
+                        ""double"": 123.456,
+                        ""string"": ""string"",
+                        ""bool"": true
+                    }
+                }
+            ";
+
+            var actual = Deserialize<Inputs>(json);
+
+            var complex = actual["complex"].ShouldBeAssignableTo<IDictionary<string, object>>();
+            complex["int"].ShouldBe(123);
+            complex["double"].ShouldBe(123.456);
+            complex["string"].ShouldBe("string");
+            complex["bool"].ShouldBe(true);
+        }
+
+        [Fact]
+        public void Deserialize_MixedValue()
+        {
+            string json = @"
+                {
+                    ""int"": 123,
+                    ""complex"": {
+                        ""int"": 123,
+                        ""double"": 123.456,
+                        ""string"": ""string"",
+                        ""bool"": true
+                    },
+                    ""bool"": true
+                }
+            ";
+
+            var actual = Deserialize<Inputs>(json);
+
+            actual["int"].ShouldBe(123);
+            actual["bool"].ShouldBe(true);
+
+            var complex = actual["complex"].ShouldBeAssignableTo<IDictionary<string, object>>();
+            complex["int"].ShouldBe(123);
+            complex["double"].ShouldBe(123.456);
+            complex["string"].ShouldBe("string");
+            complex["bool"].ShouldBe(true);
+        }
+
+        [Fact]
+        public void Deserialize_Nested_SimpleValues()
+        {
+            string json = @"
+                {
+                    ""value1"": ""string"",
+                    ""dictionary"": {
+                        ""int"": 123,
+                        ""double"": 123.456,
+                        ""string"": ""string"",
+                        ""bool"": true
+                    },
+                    ""value2"": 123
+                }
+            ";
+
+            var actual = Deserialize<Nested>(json);
+
+            actual.Value1.ShouldBe("string");
+            actual.Value2.ShouldBe(123);
+        }
+
+        [Fact]
+        public void Serialize_SimpleValues()
+        {
+            var source = new Nested
+            {
+                Value2 = 123,
+                Value1 = null
+            };
+
+            string json = Serialize(source);
+
+            json.ShouldBeCrossPlatJson(
+                @"{
+  ""Value1"": null,
+  ""Dictionary"": null,
+  ""Value2"": 123
+}".Trim());
+        }
+
+        [Fact]
+        public void Serialize_Nested_SimpleValues()
```

Comment:
The Newtonsoft.Json converters are skipped because CanWrite is false, so the built-in functionality writes `Inputs` as an `IReadOnlyDictionary<string, object>` just fine.  So these three 'nested' tests are enabled here.